tiledb/sm/filesystem/azure.cc

```diff
@@ -419,110 +404,69 @@ Status Azure::flush_blob_direct(const URI& uri) {
     std::unique_lock<std::mutex> cache_lock(write_cache_map_lock_);
     write_cache_map_.erase(uri.to_string());
   }
-
-  return Status::Ok();
 }
 
-Status Azure::is_empty_container(const URI& uri, bool* is_empty) const {
+bool Azure::is_empty_container(const URI& uri) const {
   const auto& c = client();
-  assert(is_empty);
-
   if (!uri.is_azure()) {
     throw AzureException("URI is not an Azure URI: " + uri.to_string());
   }
 
-  std::string container_name;
-  RETURN_NOT_OK(parse_azure_uri(uri, &container_name, nullptr));
-
+  auto [container_name, blob_path] = parse_azure_uri(uri);
+  bool is_empty = false;
   ::Azure::Storage::Blobs::ListBlobsOptions options;
   options.PageSizeHint = 1;
   try {
-    *is_empty = c.GetBlobContainerClient(container_name)
-                    .ListBlobs(options)
-                    .Blobs.empty();
+    is_empty = c.GetBlobContainerClient(container_name)
```

Comment:
```suggestion
    return c.GetBlobContainerClient(container_name)
```

And remove the `is_empty` variable.